pkg/collector/corechecks/servicediscovery/usm/spring.go

```diff
@@ -318,7 +318,7 @@ func (s springBootParser) GetSpringBootAppName(jarname string) (string, bool) {
 	combined := &props.Combined{Sources: []props.PropertyGetter{
 		newArgumentSource(s.ctx.args, "--"),
 		newArgumentSource(s.ctx.args, "-D"),
-		newEnvironmentSource(s.ctx.envs),
+		newEnvironmentSource(s.ctx.envs.GetVars()),
```

Comment:
This should not be using `GetVars`, the environment source should use the `Get` function from the type `EnvironmentVariables`, otherwise you don't get the benefit of having the test cases automatically catch cases where new variables are added to the parsing code but not added to the allowlist.

Reply:
Ive changed it to use envs.Get()